lib/src/screens/detail_screen/detail_screen.dart

```diff
@@ -11,89 +11,98 @@ import '/../src/constants/string_constants.dart';
-                              state.coinDetailScreen!.circulatingSupply,
-                          totalSupply: state.coinDetailScreen!.totalSupply,
-                          maxSupply: state.coinDetailScreen!.maxSupply,
-                        ),
+      ),
+      body: BlocBuilder<DetailBloc, DetailState>(
+        builder: (context, state) {
+          if (state is DetailLoadFailure) {
+            return Container(
+              color: Colors.red,
+              alignment: Alignment.center,
+              child: Text(state.errorMessage!),
+            );
+          }
+          if (state is DetailLoadInProgress) {
+            return const Center(
+              child: CircularProgressIndicator(),
+            );
+          }
+          if (state is DetailLoadSuccess) {
+            return Container(
+              decoration: const BoxDecoration(
+                  gradient: ColorConstants.backgroundGradient),
+              child: Padding(
+                padding: const EdgeInsets.symmetric(horizontal: 5.0),
+                child: Column(
+                  children: [
+                    Expanded(child: Container()),
+                    Expanded(
+                      flex: 8,
+                      child: CustomCardDetail(
+                        id: state.coinDetail.id,
+                        symbol: state.coinDetail.symbol,
+                        name: state.coinDetail.name,
+                        image: state.coinDetail.image,
+                        description: state.coinDetail.description,
+                        currentPrice: num.tryParse(percentageFormat
+                                .format(state.coinDetail.currentPrice)) ??
+                            0,
+                        priceChangePercentage24H: num.tryParse(
+                                percentageFormat.format(state
+                                    .coinDetail.priceChangePercentage24H)) ??
+                            0,
+                        high24H: num.tryParse(percentageFormat
+                                .format(state.coinDetail.high24H)) ??
+                            0,
+                        low24H: num.tryParse(percentageFormat
+                                .format(state.coinDetail.low24H)) ??
+                            0,
+                        marketCap: num.tryParse(percentageFormat
+                                .format(state.coinDetail.marketCap)) ??
+                            0,
+                        circulatingSupply: num.tryParse(percentageFormat
+                                .format(state.coinDetail.circulatingSupply)) ??
+                            0,
+                        totalSupply: num.tryParse(percentageFormat
+                                .format(state.coinDetail.totalSupply)) ??
+                            0,
+                        maxSupply: num.tryParse(percentageFormat
+                                .format(state.coinDetail.maxSupply)) ??
+                            0,
```

Comment:
We should replace `0` with something that indicates that the field was not applicable.

Reply:
I fixed it.